### PR TITLE
EN-66367: Make datasets' resource name required and unique

### PR DESCRIPTION
**Pull request by rjmac**

### CODE REVIEW

File: coordinatorlib/src/main/scala/com/socrata/datacoordinator/truth/loader/sql/messages/FromProtobuf.scala [hunk 1]
```diff
@@ -55,7 +55,7 @@ object FromProtobuf {
        nextCounterValue = di.nextCounterValue,
        localeName = di.localeName,
        obfuscationKey = di.obfuscationKey.toByteArray,
-       resourceName = di.resourceName
+       resourceName = DatasetResourceName(di.resourceName.getOrElse(sys.error("Dataset info without resource name")))
```
Comment (saraid) [inline coordinatorlib/src/main/scala/com/socrata/datacoordinator/truth/loader/sql/messages/FromProtobuf.scala:58]: Does `sys.error` explode? Or at least not return a String?

Comment (rjmac) [reply coordinatorlib/src/main/scala/com/socrata/datacoordinator/truth/loader/sql/messages/FromProtobuf.scala:58]: Yes, it explodes.

**Review by saraid — APPROVED** [review]
I'm a little baffled by the ProtoBuf interface, but I assume the compiler would have caught an egregious issue, so I'm approving.

**Comment by rjmac** [comment]
The thing with the protobuf interface is that I'm not sure you can change a field from "optional" to "required" backwards-compatibly, and I don't want to invalidate existing log entries, so I'm just leaving it optional in the protobuf spec.  Not super happy, but at least the weirdness is confined to ToProtobuf/FromProtobuf.

**Pull request merged**